packages/@aws-cdk/aws-lambda/lib/layers.ts

```diff
@@ -0,0 +1,237 @@
+import cdk = require('@aws-cdk/cdk');
+import { Code } from './code';
+import { Runtime } from './runtime';
+
+export interface LayerVersionProps {
+  /**
+   * The runtimes that this layer is compatible with.
+   *
+   * @default All runtimes are supported
+   */
+  compatibleRuntimes?: Runtime[];
+
+  /**
+   * The content of this Layer. Using the *inline* code is not permitted.
+   */
+  code: Code;
+
+  /**
+   * The description the this Lambda Layer.
+   */
+  description?: string;
+
+  /**
+   * The SPDX licence identifier or URL to the license file for this layer.
+   */
+  license?: string;
+
+  /**
+   * The name of the layer.
+   * @default a name will be generated.
+   */
+  name?: string;
+}
+
+export interface ILayerVersion {
+  /**
+   * The ARN of the Lambda Layer version that this Layer defines.
+   */
+  readonly layerVersionArn: string;
+
+  /**
+   * The runtimes compatible with this Layer.
+   */
+  readonly compatibleRuntimes?: Runtime[];
+
+  /**
+   * Grants usage of this layer to specific entities. Usage within the same account where the layer is defined is always
+   * allowed and does not require calling this method. Note that the principal that creates the Lambda function using
+   * the layer (for example, a CloudFormation changeset execution role) also needs to have the
+   * ``lambda:GetLayerVersion`` permission on the layer version.
+   *
+   * @param grantee the identification of the grantee.
+   * @param grantId the ID of the grant in the construct tree.
+   */
+  grantUsage(grantee: LayerVersionUsageGrantee, grantId: string): ILayerVersion
+}
+
+/**
+ * A reference to a Lambda Layer version.
+ */
+export abstract class LayerVersionBase extends cdk.Construct implements ILayerVersion {
+  public abstract readonly layerVersionArn: string;
+  public abstract readonly compatibleRuntimes?: Runtime[];
+
+  public grantUsage(grantee: LayerVersionUsageGrantee, grantId: string): ILayerVersion {
+    if (grantee.organizationId != null && grantee.accountId !== '*') {
+      throw new Error(`OrganizationId can only be specified if AwsAccountId is '*', but it is ${grantee.accountId}`);
```

Comment:
How about if you specify organizationId and accountId is not specified, we assume it to be '*'?

In the spirit of CDK, if it's unambiguous what you're trying to do, why should we get in the way with niggly specification details?

Reply:
Uh yeah - obviously yes.